extensions/amp-analytics/0.1/visibility-impl.js

```diff
@@ -238,18 +239,20 @@ export class Visibility {
    * @param {!VisibilityListenerCallbackDef} callback
    */
   listenOnce(config, callback) {
-    const element = this.win_.document.getElementById(config['selector']
-        .slice(1));
+    const spec = config['visibilitySpec'];
```

Comment:
Once you make the changes I mention in instrumentation.js, you will not have to make any changes to visibility-impl.js